pkg/didcomm/packager/packager.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package packager
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/btcsuite/btcutil/base58"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/transport"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/packer"
+)
+
+// Provider contains dependencies for the base packager and is typically created by using aries.Context()
+type Provider interface {
+	InboundPackers() []packer.Packer
+	Packer() packer.Packer
+}
+
+// Creator method to create new packager service
+type Creator func(prov Provider) (transport.Packager, error)
+
+// Packager is the basic implementation of Packager
+type Packager struct {
+	packer         packer.Packer
+	inboundPackers map[string]packer.Packer
```

Comment:
why `inbound`Packers? it's just a list of Packers

a Packer does pack/unpack, regardless of the direction of the message. 

any reason for the direction word in the variable `inoundPackers`?

Reply:
The packer seems to be the only argument doing both pack and unpack.

The inbound packers seems to be unpack only. Perhaps a better name would be unpackers :).

Reply:
the map of packers should not be meant for `inbound` only, during a connection, if Agent A talks with B in JWE compliant and Agent C in legacy packer, then Agent A should use the right `outbound` Packer for each connection (not just for replying to incoming messages). The the map of Packers is not meant to be for `inbound` only (Unpack()) calls.

also, `packer` field above was agreed to be the DefaultPacker which is mean to be used for new connections. For pre-existing/saved connections, the agent should store the corresponding Packer to be used for all subsequent calls (outbound and inbound).

so the word `inbound` in `inboundPacker` does not correctly represent the map of Packers as they should not be used for inbound calls only.

Reply:
Default doesn’t seem like a good prefix  -if we use that then there is a default default packer set by the framework package if the option is not provided.